packages/ui/src/components/ConfirmModal/AcModal.tsx

```diff
@@ -37,8 +38,23 @@ const AcModal = ({
         </div>
         <div className={styles.content}>{children}</div>
         <div className={styles.footer}>
-          <Button title={cancelText} type="outline" size="small" onClick={onClose} />
-          <Button title={confirmText} size="small" onClick={onConfirm ?? onClose} />
+          <Button
+            title={cancelText}
+            type="outline"
+            size="small"
+            onClick={() => {
+              onClose();
+            }}
+          />
+          {type === 'confirm' && (
+            <Button
+              title={confirmText}
+              size="small"
+              onClick={() => {
+                (onConfirm ?? onClose)();
+              }}
```

Comment:
Why do we change the usage of `onClick` handler? Seems the old pattern is more efficient

Reply:
avoid the default onClick event being passed to the callback method. Since asked,  Let me check if there is a more proper way.

Reply:
changed the handle logic to the upper scope ConfirmModalProvider